src/ocamlorg/lib/package.ml

```diff
@@ -135,6 +136,9 @@ module Info = struct
             (OpamPackage.versions_of_name pkgs name)
             versions
         in
+        let latest =
+          OpamPackage.create name (OpamPackage.Version.Map.max_binding versions)
```

Comment:
`versions` is a wrong type to put but I don't know what to put it instead.

Reply:
Yep, if we look at the error coming from the compiler we see:

```
File "src/ocamlorg/lib/package.ml", line 140, characters 71-79:
140 |           OpamPackage.create name (OpamPackage.Version.Map.max_binding versions)
                                                                             ^^^^^^^^
Error: This expression has type version_set
       but an expression was expected of type 'a Version.Map.t
```

The type `version_set` is just an alias to `Version.Set.t` (see https://v3.ocaml.org/p/opam-format/2.1.0/doc/OpamTypes/index.html#type-version_set). So we need to find the latest version from a set of versions not a map. If we then look at `OpamPackage.Version.Set` for an appropriate function for this, a little digging gives [OpamPackage.Version.Set.max_elt](https://v3.ocaml.org/p/opam-format/2.1.0/doc/OpamPackage/Version/Set/index.html#val-max_elt).

Also note that you have changed the type of `rev_deps` to be `(OpamPackage.Name.t * string option * OpamPackage.Version.t) list` where I assume the third element is the latest version which makes sense but if you were to use this value of `latest` it is a `OpamPackage.t` because you create one, you shouldn't need to create one.

Reply:
I see, I was sure about to put `OpamPackage.Version.Set.max_elt` but then since I put the type t of `(OpamPackage.Name.t * string option * OpamPackage.Version.t) list`, the compiler said we needed 'Version_map.key' so I got messed up :sweat_smile: